.circleci/config.yml

```diff
@@ -0,0 +1,235 @@
+          name: Run "black" linter
+          command: |
+            source ~/venv/bin/activate
+            black --check marsha/ 2>&1 | grep -Ev 'good job|All done'
+
+  linter_flake8:
+    <<: *python_only_config
+    steps:
+      - *restore_repo
+      - *restore_venv
+      - run:
+          name: Run "flake8" linter
+          command: |
+            source ~/venv/bin/activate
+            flake8
+
+  linter_pylint:
+    <<: *python_only_config
+    steps:
+      - *restore_repo
+      - *restore_venv
+      - run:
+          name: Run "pylint" linter
+          command: |
+            source ~/venv/bin/activate
+            pylint marsha
+
+  # django checks
+  django_checks:
+    <<: *python_only_config
+    steps:
+      - *restore_repo
+      - *restore_venv
+      - run:
+          name: Django checks
+          command: |
+            source ~/venv/bin/activate
+            django-admin.py check
+
+  # check migrations files are ok
+  check_migrations:
+    <<: *python_only_config
+    steps:
+      - *restore_repo
+      - *restore_venv
+      - run:
+          name: Check migrations
+          command: |
+            source ~/venv/bin/activate
+            django-admin.py makemigrations --check --dry-run
+
+
+  # will build the python package, using the tag as the base version, suffixed with info from git describe if not a tag
+  build_python_package:
+    <<: *python_only_config
+    steps:
+      - *restore_repo
+      - *restore_venv
+      - run:
+          name: Build package
+          command: |
+            sed -i -e "s/^version = .*$/version = $(git describe --tags)/" setup.cfg
+            source ~/venv/bin/activate
+            python setup.py sdist
+            git checkout -- setup.cfg
+      - store_artifacts:
+          path: dist/
+
+  # will build the documentation
+  build_doc:
+    <<: *python_only_config
+    steps:
+      - *restore_repo
+      - *restore_venv
+      - run:
+          name: Build package
+          command: |
+            source ~/venv/bin/activate
+            cd docs
+            sphinx-apidoc -f -o source/ ../marsha/
+            make html
+      - store_artifacts:
+          path: docs/_build
+
+workflows:
+  version: 2
+
+  marsha:
+    jobs:
+      - checkout_code
+      - install_code:
+          requires:
+            - checkout_code
+      - linter_mypy:
+          requires:
+            - install_code
+      - linter_black:
+          requires:
+            - install_code
+      - linter_flake8:
+          requires:
+            - install_code
+      - linter_pylint:
+          requires:
+            - install_code
+      - check_migrations:
+          requires:
+            - install_code
+      - django_checks:
+          requires:
+            - install_code
+      - build_python_package:
+          requires:
+            - linter_mypy
+            - linter_black
+            - linter_flake8
+            - linter_pylint
+            - check_migrations
+            - django_checks
```

Comment:
You should add a filter here to build package only for new releases. See this example: https://github.com/openfun/richie/blob/master/.circleci/config.yml#L559

Reply:
Did you read my comment above? I really want to check that the whole code is correct. So trying to build the package is a good thing. If I only try to build it when the code is finally merged and tagged and the build doesn't work, we are in a wrong state

Reply:
Okay then. Remember that we had concurrent reviews with @sampaccoud.